File: C#/HackerRank/AngryProfessor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HackerRank
{
    class AngryProfessor : IAlgorithm
    {
        private int numberOfStudents, requiredStudents;
        private List<int> arrivalTimes;
        private String result;
        

        public AngryProfessor(int numberOfStudents, int requiredStudents, List<int> arrivalTimes)
        {
            this.result = "YES";
            this.numberOfStudents = numberOfStudents;
            this.requiredStudents = requiredStudents;
            this.arrivalTimes = arrivalTimes;
        }


        public void Resolve()
        {
            int count = 0;
            foreach (int time in arrivalTimes)
            {
                count += time <= 0 ? 1 : 0;
                if (count >= this.requiredStudents)
                {
                    this.result = "NO";
                    break;
                }
            }
            
        }

        public string GetResult()
        {
            return this.result;
        }
    }
}

File: Swift/HackerRank/AngryProfessor.swift
//
//  AngryProfessor.swift
//  HackerRank
//
//  Created by Bastien on 14/07/2015.
//  Copyright (c) 2015 Bastien. All rights reserved.
//

import Foundation

enum RESPONSE : String {
    case YES = "YES"
    case NO = "NO"
    
    func getValue() -> String { return self.rawValue }
}

class AngryProfessor : AlgorithmProtocol {
    
    var result: String
    let numberOfStudent : Int, minimumStudentRequired : Int
    let arrivalTimes : Array<Int>
    
    init(numberOfStudent : Int, minimumStudentRequired: Int, arrivalTimes : Array<Int>) {
        self.result = RESPONSE.YES.getValue()
        self.numberOfStudent = numberOfStudent
        self.minimumStudentRequired = minimumStudentRequired
        self.arrivalTimes = arrivalTimes
    }
    
    func resolve() {
        var count = 0
        for arrival in self.arrivalTimes {
            if (arrival <= 0) {
                count++
            }
        }
        self.result = count >= self.minimumStudentRequired ? RESPONSE.NO.getValue() : RESPONSE.YES.getValue()
    }
    
     func getResult() -> String {
        return self.result;
    }
}